### ISSUE TITLE
Coding

### ISSUE THREAD

**Issue by shinfang** [GentleDell/imu_preintegration_Cpp#1]
I'm currently implementing IMU Preintegration theory in my coding based on your coding. 
Would like to ask if you've tested you coding and make sure it's working before?

**Comment by GentleDell** [comment]
Yes, I have tested the preintegration but the Jacobian and error function have not been tested yet.

**Comment by Lachiven** [comment]
您好，想问下您最终测试Jacobian和噪声矩阵了么，不知道这两个怎么使用，能交流一下么？1353632881@qq.com。另外因子图的加入和手推的预积分是作比较用的么？因子图的库好像带了预积分的功能？

**Comment by GentleDell** [comment]
No, I did not test them. I haven't worked on this project for 4 years (since 2017) and there is plan to update it. 

I would propose you to read [this paper](http://rpg.ifi.uzh.ch/docs/TRO16_forster.pdf) for more details about how to use the Jacobian and the variance matrix.

Yes, I added them for comparison.

The factor graph provides a good sample of how to do IMU preintegration and you can follow the sample to build your own application. I used it 3 years ago and it worked pretty good. 

Hope this helps.
